### PR TITLE
Week4 homework Abdulrahman sadek

### PR DESCRIPTION
**Pull request by Abdulrahmansadek**

### CODE REVIEW

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 1]
```diff
@@ -11,13 +11,9 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
...
 // ! Function to be tested
 function doubleEvenNumbers(numbers) {
-  // TODO rewrite the function body using `map` and `filter`.
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
+  const newNumbers = numbers
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:14]: Well done!

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 2]
```diff
@@ -31,8 +31,10 @@ const mondayTasks = [
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function computeEarnings(tasks, rate) {
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:34]: Well done!
I'd suggest to use even more descriptive names for the variables. Can you rethink `monday` and `calculateRate` variable names so they reflect better what they are storing?

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 3]
```diff
@@ -26,11 +26,16 @@ const fruitBasket = [
...
+  return filterBasket;
 }
 
-// ! Unit tests (using Jest)
+sanitizeFruitBasket(fruitBasket, 'lemon');
+
+// // ! Unit tests (using Jest)
 describe('sanitizeFruitBasket', () => {
   test('should take two parameters', () => {
     // TODO replace next line with your code
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:41]: I think the tests are missing, you also have to make them work.

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js [hunk 4]
```diff
@@ -17,10 +17,12 @@ function createObservable() {
   const subscribers = [];
   return {
     subscribe: function (subscriber) {
-      // TODO complete this function
+      subscribers.push(subscriber);
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:20]: Well done!

File: 1-JavaScript/Week4/homework/ex5-wallet/index.js [hunk 5]
```diff
@@ -26,9 +26,9 @@ function createWallet(name, cash = 0) {
 
   function transferInto(wallet, amount) {
     console.log(
-      `Transferring ${eurosFormatter.format(amount)} from ${name} to ${
-        wallet.name
-      }`
+      `Transferring ${eurosFormatter.format(
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex5-wallet/index.js:29]: Well done!

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js (hunk 3 above)
Comment (Abdulrahmansadek) [reply 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:41]: Hey @killoncio thanks for reviewing my homework i will work on the test.

Comment (robvk) [reply 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:41]: Hi @Abdulrahmansadek , looks like you haven't done this yet?

Comment (Abdulrahmansadek) [reply 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:41]: hi @robvk  yes im will be sure to get them done by today . sorry I was so busy with the project .

**Review by Abdulrahmansadek — COMMENTED** [review]
@robvk  here also resolving the issue of the fruits basket ex.

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 6]
```diff
@@ -35,22 +35,22 @@ function sanitizeFruitBasket(basket, fruits) {
...
+    expect(sanitizeFruitBasket).toHaveLength(2);
   });
 
   test('should not modify the original `fruitBasket` array', () => {
     // Save the original contents of the fruit basket
     const originalFruitBasketContents = [...fruitBasket];
-    // TODO replace next line with your code
-    expect(false).toBe(true);
+    sanitizeFruitBasket(fruitBasket, 'lemon');
+    expect(originalFruitBasketContents).toBe(fruitBasket);
```
Comment (robvk) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:48]: Does this do the ```should not modify the original `fruitBasket` array``` check?

**Review by robvk — CHANGES_REQUESTED** [review]
Not yet, you made a mistake with the should not modify check

**Review by Abdulrahmansadek — COMMENTED** [review]
hi @robvk  I think now is working well .

**Pull request closed without merging**